ipaserver/install/dsinstance.py

```diff
@@ -1032,22 +1032,52 @@ def uninstall(self):
                 logger.error(
                     'Unable to restart DS instance %s: %s', ds_instance, e)
 
+    def get_server_cert_nickname(self, serverid=None):
+        """
+        Retrieve the nickname of the server cert used by dirsrv.
+
+        The method reads the NSS DB instead of using the attribute
+        nssslpersonalityssl of cn=RSA,cn=encryption,cn=config because
+        LDAP is not always accessible when we need to get the nickname
+        (for instance during uninstall).
+        """
+        if serverid is None:
+            serverid = self.get_state("serverid")
+        if serverid is not None:
+            dirname = config_dirname(serverid)
+            try:
+                dsdb = certs.CertDB(self.realm, nssdir=dirname)
+                nickname = dsdb.find_server_certs()[0][0]
+                return nickname
+            except Exception as e:
+                logger.error("Unable to find server cert nickname in %s (%s)",
+                             dirname, e)
```

Comment:
This will probably print "index out of range" in case no server certs were found. Not sure whether we want that, although I could live with it.